src/test/java/io/orkes/conductor/client/e2e/JDBCWorkerTests.java

```diff
@@ -0,0 +1,347 @@
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+public class JDBCWorkerTests {
+    private static WorkflowClient workflowClient;
+    private static MetadataClient metadataClient;
+    private static final String WORKFLOW_NAME = RandomStringUtils.randomAlphanumeric(5);
+    private final static String TABLE_NAME = RandomStringUtils.randomAlphabetic(5).toLowerCase();
+
+    @BeforeAll
+    public static void init() {
+        ApiClient apiClient = ApiUtil.getApiClientWithCredentials();
+        workflowClient = new OrkesWorkflowClient(apiClient);
+        metadataClient = new OrkesMetadataClient(apiClient);
+
+        registerWorkflowDef();
+    }
+
+    @AfterAll
+    public static void clearResource() {
+        SearchResult<WorkflowSummary> found = workflowClient.search("workflowType IN (" + WORKFLOW_NAME + ")");
+        found.getResults().forEach(workflowSummary -> {
+            try {
+                workflowClient.deleteWorkflow(workflowSummary.getWorkflowId(), false);
+                System.out.println("Going to delete " + workflowSummary.getWorkflowId());
+            } catch (Exception e) {
+            }
+        });
+    }
+
+    @BeforeEach
+    public void initializeDatabase() {
+        createTable();
+        insertSeedDataInTable();
+    }
+
+    @AfterEach
+    public void clearData() {
+        dropTable();
+
+        SearchResult<WorkflowSummary> found = workflowClient.search("workflowType IN (" + WORKFLOW_NAME + ") AND status IN (RUNNING)");
+        found.getResults().forEach(workflowSummary -> {
+            try {
+                workflowClient.terminateWorkflow(workflowSummary.getWorkflowId(), "terminate");
+                System.out.println("Going to terminate " + workflowSummary.getWorkflowId());
+            } catch (Exception e) {
+            }
+        });
+    }
+
+    @Test
+    public void testJDBCWorkerSelect() {
+        Map<String, Object> inputMap = Map.of(
+                "connectionId", "postgres1",
+                "statement", "SELECT * FROM " + TABLE_NAME,
+                "parameters", List.of(),
+                "type", "SELECT"
+        );
+
+        StartWorkflowRequest startWorkflowRequest = new StartWorkflowRequest();
+        startWorkflowRequest.setName(WORKFLOW_NAME);
+        startWorkflowRequest.setInput(inputMap);
+
+        String workflowId = workflowClient.startWorkflow(startWorkflowRequest);
+
+        await().atMost(30, TimeUnit.SECONDS).untilAsserted(() -> {
+            Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+            assertTrue(workflow.getTasks().get(0).getStatus().isTerminal());
+        });
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        Task task = workflow.getTasks().get(0);
+
+        assertEquals(Task.Status.COMPLETED, task.getStatus());
+        assertEquals(10, ((List<Object>) task.getOutputData().get("result")).size());
```

Comment:
From where this 10 records are coming?

Reply:
Before the test, I inserted 10 records into the table, see [insertSeedDataInTable](https://github.com/orkes-io/orkes-conductor-client/blob/b35b9096d0fa35e422db71b529643439c1c2753f/src/test/java/io/orkes/conductor/client/e2e/JDBCWorkerTests.java#L329) function.